packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -510,11 +510,18 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     }
 
     // Adjust the clip rect for this sliver by the overlap from the previous sliver.
+    final double startOfOverlap = child.constraints.viewportMainAxisExtent - child.constraints.remainingPaintExtent;
+    // We tried to subtract infinity from infinity and got NaN.
+    // This can happen when a RenderShrinkWrappingViewport is given infinite
```

Comment:
Can you add a little more details here when this can happen? Somehwere along the lines: When a RenderShrinkWrappingViewport itself is inside another viewport it basically turns into a Column/Row, which has infinite space in the horizontal/vertical direction. This results in an infinite viewportMainAxisExtent. In that case, we can just return the size of the viewport (viewportClip) because the parent viewport will take care of the actual clipping.

Reply:
Tried to clarify a bit more  but before I saw this.  LMK if that works or if I should update further.